### PR TITLE
Refactor E2E tests to use wait instead of fixed sleep

### PR DESCRIPTION
**Pull request by tcar121293**
## Description
Refactor E2E tests to use wait instead of fixed sleep
## Related Issue Or Context

Closes: #175 

## How Has This Been Tested? Testing details.
Run e2e tests locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: e2e/evm/test.go [hunk 1]
```diff
@@ -213,10 +222,57 @@ func (s *IntegrationTestSuite) TestGenericDeposit() {
...
-	time.Sleep(120 * time.Second)
-
+	WaitForProposalExecuted(s, s.bridgeAddr2)
 	// Asset hash sent is stored in centrifuge asset store contract
 	exists, err := assetStoreContract2.IsCentrifugeAssetStored(hash)
 	s.Nil(err)
 	s.Equal(true, exists)
 }
+
+func WaitForProposalExecuted(s *IntegrationTestSuite, bridge common.Address) {
```
Comment (mpetrun5) [inline e2e/evm/test.go:232]: Probably can be moved to some utility file and accept client.
Don't think IntegrationTestSuite param is needed.

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: chains/evm/cli/utils/utils.go [hunk 2]
```diff
@@ -25,25 +23,6 @@ func init() {
...
-//	return ProposalStatus(status) == Passed
-//}
-//
-//func IsExecuted(status uint8) bool {
-//	return ProposalStatus(status) == Executed
-//}
-
 // UserAmountToWei converts decimal user friendly representation of token amount to 'Wei' representation with provided amount of decimal places
 // eg UserAmountToWei(1, 5) => 100000
 func UserAmountToWei(amount string, decimal *big.Int) (*big.Int, error) {
```
Comment (MakMuftic) [inline chains/evm/cli/utils/utils.go:28]: Could you also move these functions to some separate file - maybe together with ` WaitForProposalExecuted`, as they don't make any sense inside this file.

**Review by MakMuftic — CHANGES_REQUESTED** [review]

File: e2e/evm/util.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Unsubscribe()
+
+	a, err := abi.JSON(strings.NewReader(consts.BridgeABI))
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("dosassss\nnn")
```
Comment (mpetrun5) [inline e2e/evm/util.go:40]: ?

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

**Comment by MakMuftic** [comment]
@tcar121293 run mock generation locally, it is failing on CI

File: e2e/evm/test.go [hunk 4]
```diff
@@ -2,32 +2,38 @@ package evm
...
 	"github.com/ChainSafe/chainbridge-core/keystore"
+	"github.com/ethereum/go-ethereum"
 
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/rs/zerolog/log"
 	"github.com/stretchr/testify/suite"
 )
 
+var TestTimeout = time.Second * 600
```
Comment (MakMuftic) [inline e2e/evm/test.go:30]: I would move this together with `WaitForProposalExecuted` as it is used there.

File: e2e/evm/util.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+	a, err := abi.JSON(strings.NewReader(consts.BridgeABI))
+	if err != nil {
+		panic(err)
+	}
+	timeout := time.After(TestTimeout)
+	for {
+		select {
+		case evt := <-ch:
+			fmt.Println("nesjedosloooooo\nnn")
```
Comment (MakMuftic) [inline e2e/evm/util.go:44]: clean up

File: e2e/evm/util.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+	sub, err := client.SubscribeFilterLogs(context.Background(), query, ch)
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Unsubscribe()
+
+	a, err := abi.JSON(strings.NewReader(consts.BridgeABI))
+	if err != nil {
+		panic(err)
```
Comment (MakMuftic) [inline e2e/evm/util.go:38]: I think it would be better to return this error and then decide in setup itself what to do if the error occurs

**Review by MakMuftic — CHANGES_REQUESTED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**